pkg/controller/currency-controller.go

```diff
@@ -7,18 +7,22 @@ import (
 	"github.com/gin-gonic/gin"
 )
 
-type CurrencyController struct {
-	currencyService *service.CurrencyService
+type CurrencyController interface {
+	GetLatest(ctx *gin.Context)
```

Comment:
You don't need to define controller interfaces, because the direction of dependencies is not broken here: the web framework calls controllers; controllers call services and return the result.

Controllers do not depend on services because there are interfaces. But I don't know about the dependence of a web framework on controllers. A web controller is a framework-specified object that must receive certain arguments and return a framework-defined result. You can define independent controllers that receive and return their own objects, but I don't see the point in this, because the controller should be a simple entry point object without complex logic, so this solution only complicates the project structure.

Reply:
This comment also applies to other controllers

Reply:
Totaly agree with your comments. I made these interfaces since for testing golang requires them. But quickly realized that they are redundant for testing.
Removed these interfaces.